examples/api/lib/material/menu_bar/menu_bar.1.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Example for MenuBar.adaptive.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+void main() {
+  runApp(const MaterialApp(
+    title: 'Adaptive Menu Sample',
+    home: Material(child: Home()),
+  ));
+}
+
+enum MenuSelection {
+  edit('Edit'),
+  cut('Cut'),
+  copy('Copy'),
+  paste('Paste'),
+  file('File'),
+  open('Open'),
+  quit('Quit'),
+  save('Save'),
+  saveAs('Save As...');
+
+  const MenuSelection(this.label);
+  final String label;
+}
+
+
+class Home extends StatefulWidget {
+  const Home({super.key});
+
+  @override
+  State<Home> createState() => _HomeState();
+}
+
+class _HomeState extends State<Home> {
+  bool isPlatformMenu = false;
+  TextDirection textDirection = TextDirection.ltr;
+  bool enabled = true;
+  bool checked = false;
+
+  void _onSelected(MenuSelection item) {
+    debugPrint('Activated ${item.name}');
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final bool isMacOS = defaultTargetPlatform == TargetPlatform.macOS;
+    final bool meta = isMacOS;
+    final bool control = !meta;
```

Comment:
In an adaptive menu I wish I wouldn't have to think about this difference either... Not sure what to do about this (and if we want to do anything about it, we could do it in a follow-up). Do we maybe need an adaptive SingleActivator that correctly translates between these two? (I haven't thought this through at all, just throwing an idea out.)

Reply:
If anything, we should document that developers need to think about this even when using the adaptive constructor.